spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java

```diff
@@ -150,50 +174,72 @@ public void afterPropertiesSet() {
 		this.renewQuery = String.format(this.renewQuery, this.prefix);
 	}
 
-	@Transactional(propagation = Propagation.REQUIRES_NEW)
+	@Override
+	public void afterSingletonsInstantiated() {
+		if (this.transactionManager == null) {
+			this.transactionManager = this.applicationContext.getBean(PlatformTransactionManager.class);
```

Comment:
Maybe try/catch with a friendly message to indicate if there is not one unique TM in the context, the registry needs one to be passed in?